x/wasm/client/testutil/query.go

```diff
@@ -0,0 +1,466 @@
+				Creator:               val.Address.String(),
+				DataHash:              dataHash,
+				InstantiatePermission: types.AllowEverybody,
+			},
+		},
+		"no codeID": {
+			[]string{},
+			false,
+			nil,
+		},
+		"no exist codeID": {
+			[]string{"100"},
+			false,
+			nil,
+		},
+	}
+
+	for name, tc := range testCases {
+		tc := tc
+
+		s.Run(name, func() {
+			cmd := cli.GetCmdQueryCodeInfo()
+			out, err := clitestutil.ExecTestCLICmd(val.ClientCtx, cmd, append(tc.args, s.queryCommonArgs()...))
+			if !tc.valid {
+				s.Require().Error(err)
+				return
+			}
+			s.Require().NoError(err)
+
+			var codeInfo types.CodeInfoResponse
+			s.Require().NoError(val.ClientCtx.Codec.UnmarshalJSON(out.Bytes(), &codeInfo), out.String())
+			s.Require().Equal(tc.expected, &codeInfo)
+		})
+	}
+}
+
+func (s *IntegrationTestSuite) TestGetCmdGetContractInfo() {
+	val := s.network.Validators[0]
+
+	codeID, err := strconv.ParseUint(s.codeID, 10, 64)
+	s.Require().NoError(err)
+
+	testCases := map[string]struct {
+		args     []string
+		valid    bool
+		expected proto.Message
+	}{
+		"valid query": {
+			[]string{
+				s.contractAddress,
+			},
+			true,
+			&types.QueryContractInfoResponse{
+				Address: s.contractAddress,
+				ContractInfo: types.ContractInfo{
+					CodeID:    codeID,
+					Creator:   val.Address.String(),
+					Admin:     val.Address.String(),
+					Label:     "TestContract",
+					Created:   nil,
+					IBCPortID: "",
+					Extension: nil,
+				},
+			},
+		},
+		"no contractAddress": {
+			[]string{},
+			false,
+			nil,
+		},
+		"wrong contactAddress": {
+			[]string{
+				"abc",
+			},
+			false,
+			nil,
+		},
+	}
+
+	for name, tc := range testCases {
+		tc := tc
+
+		s.Run(name, func() {
+			cmd := cli.GetCmdGetContractInfo()
+			out, err := clitestutil.ExecTestCLICmd(val.ClientCtx, cmd, append(tc.args, s.queryCommonArgs()...))
+			if !tc.valid {
+				s.Require().Error(err)
+				return
+			}
+			s.Require().NoError(err)
+
+			var contractInfo types.QueryContractInfoResponse
+			s.Require().NoError(val.ClientCtx.Codec.UnmarshalJSON(out.Bytes(), &contractInfo), out.String())
+			s.Require().Equal(tc.expected, &contractInfo)
+		})
+	}
+}
+
+func (s *IntegrationTestSuite) TestGetCmdGetContractStateAll() {
+	val := s.network.Validators[0]
+
+	testCases := map[string]struct {
+		args     []string
+		valid    bool
+		expected proto.Message
+	}{
+		"valid query": {
+			[]string{
+				s.contractAddress,
+			},
+			true,
+			&types.QueryAllContractStateResponse{
+				Models: []types.Model{
+					{
+						Key:   []byte("config"),
+						Value: []byte(fmt.Sprintf("{\"verifier\":\"%s\",\"beneficiary\":\"%s\",\"funder\":\"%s\"}", s.verifier, s.beneficiary, s.verifier)),
+					},
+				},
+				Pagination: &query.PageResponse{},
+			},
+		},
+		"wrong bech32_address": {
+			[]string{
+				"xxx",
+			},
+			false,
+			nil,
+		},
+		"no exist bech32_address": {
+			[]string{
+				"link1hmayw7vv0p3gzeh3jzwmw9xj8fy8a3kmpqgjrysljdnecqkps02qrq5rvm",
```

Comment:
The literal is also being used in other places, so you may define a variable for it. It would also make sure it is of valid format, but not in the state.

Reply:
OK, if this address is using others, I think  it need to change some other unregistered address clearly.  I'll define an address that no exist but valid type. So I am thinking of using `randomAddress`.